### PR TITLE
Updated the CRUD of reference data from v1.3

### PR DESCRIPTION
**Pull request by mickaelandrieu**
Hello, this is part of the cookbooks/tutorial to help a beginner create a reference data and be able to manage it in the back office. (see https://github.com/akeneo/pim-docs/issues/488)

Note this pull request depends on https://github.com/akeneo/pim-docs/pull/487 as we need to install CustomEntity Bundle and understand what it provides.

> Needed to be cherry picked in ``1.7``, ``1.6`` and ``1.5`` branches.

/!\ don't merge it => should be done in ``1.5`` branch

### CODE REVIEW

File: cookbook/ui_customization/create_a_reference_data_crud.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
+How to create the screens to manage a Reference Data
+====================================================
+
+.. note::
+    The code inside this cookbook entry requires you to install the :doc:`/reference/bundles/custom_entity_bundle/index`.
```
Comment (mickaelandrieu) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: this is the link to the previous pull request, when merged :)

File: cookbook/ui_customization/create_a_reference_data_crud.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                        label:     Name
+                        data_name: rd.name
+            sorters:
+                columns:
+                    code:
+                        data_name: rd.code
+                    name:
+                        data_name: rd.name
+                default:
+                    code: '%oro_datagrid.extension.orm_sorter.class%::DIRECTION_ASC'
```
Comment (mickaelandrieu) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: What is missing here is a reference documentation about grid link, see https://github.com/akeneo/pim-docs/issues/490

File: cookbook/ui_customization/create_a_reference_data_crud.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    The code inside this cookbook entry requires you to install the :doc:`/reference/bundles/custom_entity_bundle/index`.
+
+Creating the Reference Data Management Screens
+**********************************************
+
+The Grid
+--------
+
+To benefit from the grid component (which comes natively with filtering and sorting),
+you can define the vendor grid as following:
```
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: _as follows:_

File: cookbook/ui_customization/create_a_reference_data_crud.rst [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+        public function getName()
+        {
+            return 'app_enrich_color';
+        }
+    }
+
+.. note::
+
+    Want to know more about forms? Take a look at the `Symfony documentation`_.
```
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: _Want to learn more_

File: cookbook/ui_customization/create_a_reference_data_crud.rst [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+Declare the CRUD actions
+------------------------
+
+Now we have created the grid and the required form for both creation and update,
+we only need to declare the reference data as "custom entities":
```
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: as a "custom entity".

File: cookbook/ui_customization/create_a_reference_data_crud.rst [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+                edit:
+                    form_type: app_enrich_color # Identical to the form type `getName()` value
+                create:
+                    form_type: app_enrich_color
+
+.. note::
+
+    You can define the same form type for both creation and edition tasks.
+
+From this point a working grid screen should be visible at ``/app_dev.php/enrich/color``.
```
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: _At this point, a working grid should be visible at_

File: cookbook/ui_customization/create_a_reference_data_crud.rst [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+Create an entry point in Back Office
```
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: _in the Back office_

File: cookbook/ui_customization/create_a_reference_data_crud.rst [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+------------------------------------
+
+Most of the time, we want to make the screens available for our customers. Fortunately, this is
+really easy to add a new menu entry in back office:
```
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: _it is really easy to_

File: cookbook/ui_customization/create_a_reference_data_crud.rst [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+                tree:
+                application_menu:
+                    children:
+                        pim_reference_data_tab:
+                            children:
+                                app_enrich_color: ~
+
+.. note::
+
+    Want to know more about the menu management? Take a look at the :doc:`/cookbook/catalog_structure/creating_a_reference_data` cookbook.
```
Comment (MarieMinasyan) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: _learn_ instead of _know_

Comment (mickaelandrieu) [inline cookbook/ui_customization/create_a_reference_data_crud.rst]: dully noted :)

**Comment by mickaelandrieu** [comment]
@MarieMinasyan sorry, closed in favor of https://github.com/akeneo/pim-docs/pull/494

Thanks for review, I've addressed your comments and completed my pull request

**Pull request closed without merging**